ciao-launcher/instance.go

```diff
@@ -393,7 +362,8 @@ DONE:
 			id.statsTimer = nil
 			id.ovsCh <- &ovsStateChange{id.instance, ovsStopped}
 			id.st = nil
-			id.unmapVolumes()
+			killMe(id.instance, false, true, id.doneCh, id.ac, &id.instanceWg)
```

Comment:
This piece of code gets called when the VM shuts down.  Pre-migration the instance continues to reside on the compute node even though it is no longer running, so the instance loop keeps running.

With migration instances that shutdown or just finish executing are automatically evacuated.  Thats what the call to killMe does.  It basically inserts a SSNTP DELETE command into launcher's command queue so the instance's state gets deleted.  When this DELETE command is processed by the instance go routine in the deleteCommand function the volumes are unmapped so there is no longer any need to unmap them here.